### PR TITLE
fix: RoaringInt `can_compress` was erroneously always `None`

### PR DESCRIPTION
**Pull request by lwwmanning**
(unless a user produced a RoaringIntArray manually, in which case it was Some... but a no-op)

### CODE REVIEW

**Comment by lwwmanning** [comment]
split out of #1000

**Review by robert3005 — APPROVED** [review]

**Comment by robert3005** [comment]
Now we wait for the fuzzer for find all the bugs with it

File: encodings/roaring/src/integer/mod.rs [hunk 1]
```diff
@@ -94,17 +114,41 @@ impl ArrayValidity for RoaringIntArray {
 
 impl IntoCanonical for RoaringIntArray {
     fn into_canonical(self) -> VortexResult<Canonical> {
-        todo!()
+        try_cast(
+            PrimitiveArray::from_vec(self.bitmap().to_vec(), Validity::NonNullable),
+            self.dtype(),
+        )
+        .and_then(|a| a.into_primitive())
```
Comment (robert3005) [inline encodings/roaring/src/integer/mod.rs]: a.into_canonical() does the right thing, no need to unwrap it

File: encodings/roaring/src/integer/mod.rs [hunk 2]
```diff
@@ -94,17 +114,41 @@ impl ArrayValidity for RoaringIntArray {
...
+                .buffer()
+                .vortex_expect("Missing buffer in RoaringIntArray"),
+        )
     }
 }
 
-impl ArrayStatisticsCompute for RoaringIntArray {}
+impl ArrayStatisticsCompute for RoaringIntArray {
+    fn compute_statistics(&self, stat: Stat) -> VortexResult<StatsSet> {
+        let mut stats = self.statistics().to_set();
```
Comment (robert3005) [inline encodings/roaring/src/integer/mod.rs]: This is already checked before being called, i.e. you will never be asked to compute stat that already exists in the set

File: encodings/roaring/src/integer/mod.rs [hunk 3]
```diff
@@ -94,17 +114,41 @@ impl ArrayValidity for RoaringIntArray {
...
+            return Ok(stats);
+        }
+
+        if stat == Stat::TrailingZeroFreq || stat == Stat::BitWidthFreq || stat == Stat::RunCount {
+            let primitive = PrimitiveArray::from_vec(self.bitmap().to_vec(), Validity::NonNullable);
+            let prim_stats = primitive.statistics().to_set();
+            stats.merge(&prim_stats);
+        }
+
+        Ok(stats)
```
Comment (robert3005) [inline encodings/roaring/src/integer/mod.rs]: You could also say we don't support these

**Pull request merged**